app/scripts/modules/core/src/presentation/main.less

```diff
@@ -641,6 +641,11 @@ body > .container {
   }
 }
 
+/* This class is used to override the uibModal default z-index value of 1050 */
+.modal-z-index-fix {
+  z-index: var(--layer-critical) !important;
+}
+
 .modal-pages {
   width: 1200px;
   .modal-page {
```

Comment:
Did you try adding `z-index: var(--layer-critical)`  under `modal-page`? I think that may do the trick, then you can remove the rest of the code.

Reply:
Yup, tried adding that, but it was still being overridden by the z-index defined in uibModal from Angular's bootstrap UI.